File: layers/src/main/scala/stasis/layers/telemetry/analytics/AnalyticsClient.scala
package stasis.layers.telemetry.analytics

import scala.concurrent.Future

import org.apache.pekko.Done

trait AnalyticsClient {
  def sendAnalyticsEntry(entry: AnalyticsEntry): Future[Done]
}

object AnalyticsClient {
  trait Provider {
    def client: AnalyticsClient
  }

  object Provider {
    def apply(f: () => AnalyticsClient): Provider = new Provider {
      override def client: AnalyticsClient = f()
    }
  }
}

File: client-ios/lib/Sources/StasisClientLib/Telemetry/Analytics/AnalyticsClient.swift
public protocol AnalyticsClient: Sendable {
    func sendAnalyticsEntry(_ entry: AnalyticsEntry) async throws
}

public struct NoOpAnalyticsClient: AnalyticsClient {
    public init() {}
    public func sendAnalyticsEntry(_ entry: AnalyticsEntry) async throws {}
}
